pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1404,17 +1411,34 @@ func buildGRPCDestination(
 	return res, invalidBackendErr, nil
 }
 
-func buildDestination(ctx configContext, to k8s.BackendRef, ns string, enforceRefGrant bool) (*istio.Destination, *ConfigError) {
```

Comment:
Id rather just make buildDestination take in a gvk, seems much simpler and would make this a ~5 line change?

Reply:
This is indeed what I was doing in my initial attempt to fix it, which resulted in a call to `buildTCPDestination(... ,gvk.TLSRoute)`, which can look a bit strange.
Also since the previous code check all Route type permission using `ReferenceGrant` for `HTTPRoute`, I didn't have any more context to determine if it could use a `ReferenceGrant` for one Route instead of another, so I made the current change. 
I'll simplify the current changes.